sklearn/model_selection/_validation.py

```diff
@@ -937,25 +940,31 @@ def permutation_test_score(estimator, X, y, groups=None, cv=None,
 
     # We clone the estimator to make sure that all the folds are
     # independent, and that it is pickle-able.
-    score = _permutation_test_score(clone(estimator), X, y, groups, cv, scorer)
+    score = _permutation_test_score(clone(estimator), X, y, groups, cv, scorer,
+                                    fit_params=fit_params)
     permutation_scores = Parallel(n_jobs=n_jobs, verbose=verbose)(
         delayed(_permutation_test_score)(
             clone(estimator), X, _shuffle(y, groups, random_state),
-            groups, cv, scorer)
+            groups, cv, scorer, fit_params=fit_params)
         for _ in range(n_permutations))
     permutation_scores = np.array(permutation_scores)
     pvalue = (np.sum(permutation_scores >= score) + 1.0) / (n_permutations + 1)
     return score, permutation_scores, pvalue
 
 
-def _permutation_test_score(estimator, X, y, groups, cv, scorer):
+def _permutation_test_score(estimator, X, y, groups, cv, scorer,
+                            fit_params):
     """Auxiliary function for permutation_test_score"""
+    # Adjust length of sample weights
+    fit_params = fit_params if fit_params is not None else {}
     avg_score = []
     for train, test in cv.split(X, y, groups):
         X_train, y_train = _safe_split(estimator, X, y, train)
         X_test, y_test = _safe_split(estimator, X, y, test, train)
-        estimator.fit(X_train, y_train)
-        avg_score.append(scorer(estimator, X_test, y_test))
+        train_fit_params = {k: v[train] for k, v in fit_params.items()}
```

Comment:
Do we not use something like safe_indexing in _fit_and_score to ensure correct handling of different types? Please add a test where sample_weight is a list rather than an array.